src/category_theory/limits/concrete_category.lean

```diff
@@ -19,4 +20,64 @@ namespace category_theory.limits
 
 attribute [elementwise] cone.w limit.lift_π limit.w cocone.w colimit.ι_desc colimit.w
 
+local attribute [instance] concrete_category.has_coe_to_fun concrete_category.has_coe_to_sort
+
+section limits
+
+variables {C : Type u} [category.{v} C] [concrete_category.{v} C]
+  {J : Type v} [small_category J] (F : J ⥤ C) [preserves_limit F (forget C)]
+
+lemma concrete.to_product_injective_of_is_limit {D : cone F} (hD : is_limit D) :
+  function.injective (λ (x : D.X) (j : J), D.π.app j x) :=
+begin
+  let E := (forget C).map_cone D,
+  let hE : is_limit E := is_limit_of_preserves _ hD,
+  let G := types.limit_cone (F ⋙ forget C),
+  let hG := types.limit_cone_is_limit (F ⋙ forget C),
+  let T : E.X ≅ G.X := hE.cone_point_unique_up_to_iso hG,
+  change function.injective (T.hom ≫ (λ x j, G.π.app j x)),
+  have h : function.injective T.hom,
+  { intros a b h,
+    suffices : T.inv (T.hom a) = T.inv (T.hom b), by simpa,
+    rw h },
```

Comment:
```suggestion
  have h : function.injective T.hom,
  { rw ←mono_iff_injective,
    apply_instance },
```
I hoped this would work, but it doesn't... instance search tries to find an is_iso instance for `T.hom`, and ends up looping using `comp_is_iso` seemingly because of composing with the identity function? I think this is the second time I've seen typeclass instance look further than I expected it to in the category of types

Reply:
> Could you add the colimits version and TODOs for the other special shapes? (Of course I'm happy for you to add the other special shapes to this but that's up to you!)

I added the colimit analogues with the special cases of `wide_pushout`. I added `TODO` comments for (co)products and (co)equalizers. These other special shapes should be easy enough, but I don't want to hold off cleaning up the `for_mathlib` folder in LTE for too long with this, so I think a `TODO` comment should be fine for now.